tests/test_sftpserver.cpp

```diff
@@ -802,6 +804,68 @@ TEST_F(SftpServer, handles_reads)
     ASSERT_THAT(num_calls, Eq(1));
 }
 
+TEST_F(SftpServer, handle_extended_link)
+{
+    mpt::TempDir temp_dir;
+    auto file_name = temp_dir.path() + "/test-file";
+    auto link_name = temp_dir.path() + "/test-link";
+    make_file_with_content(file_name);
+
+    auto sftp = make_sftpserver();
+    auto msg = make_msg(SFTP_EXTENDED);
+    std::string extended_method("hardlink@openssh.com");
+    auto submessage = name_as_char_array(extended_method);
+    msg->submessage = submessage.data();
+    auto name = name_as_char_array(file_name.toStdString());
+    msg->filename = name.data();
+
+    auto target_name = name_as_char_array(link_name.toStdString());
+    REPLACE(sftp_client_message_get_data, [&target_name](auto...) { return target_name.data(); });
+
+    int num_calls{0};
+    auto reply_status = make_reply_status(msg.get(), SSH_FX_OK, num_calls);
+    REPLACE(sftp_reply_status, reply_status);
+    REPLACE(sftp_get_client_message, make_msg_handler());
+
+    sftp.run();
+
+    ASSERT_THAT(num_calls, Eq(1));
+
+    QFileInfo info(link_name);
+    EXPECT_TRUE(QFile::exists(link_name));
+    EXPECT_TRUE(content_match(link_name, "this is a test file"));
+}
+
+TEST_F(SftpServer, handle_extended_rename)
+{
+    mpt::TempDir temp_dir;
+    auto old_name = temp_dir.path() + "/test-file";
+    auto new_name = temp_dir.path() + "/test-renamed";
+    make_file_with_content(old_name);
+
+    auto sftp = make_sftpserver();
+    auto msg = make_msg(SFTP_EXTENDED);
+    std::string extended_method("posix-rename@openssh.com");
+    auto submessage = name_as_char_array(extended_method);
```

Comment:
This could just be auto submessage = name_as_char_array("posix-rename@openssh.com");